src/validator.ts

```diff
@@ -234,6 +217,290 @@ function validateWorkingInput(passedOptions?: Partial<ValidateOptions>) {
 
 }
 
+function applySpecificationOverrides() {
+
+    // extend resources specification with SAM specifics
+    if (!!~workingInputTransform.indexOf('AWS::Serverless-2016-10-31')) {
+        resourcesSpec.extendSpecification(samData.samResources20161031);
+    }
+
+    // normalize Tag property type access
+    let tagPropertyTypes: any = {};
```

Comment:
Why is this required?

Reply:
This actually creates the specific `ResourceType.Tag` property types, as depicted above.